# ファイルの確認

In [2]:
ls

README.md
mopacの実行.ipynb
mopacの計算ファイルの確認.ipynb
openbabel及びpybelの使い方.ipynb
phenol.arc
phenol.dat
phenol.out
read_file.py


# outファイルの読み取り

In [6]:
with open('phenol.out', mode='r') as f:
    lines = f.readlines()
    for line in lines:
        print(line, end='')

 *******************************************************************************
 **                                                                           **
 **                              MOPAC v22.0.4                                **
 **                                                                           **
 *******************************************************************************
 **          Digital Object Identifier (DOI): 10.5281/zenodo.6511958          **
 **    Visit the DOI location for information on how to cite this program.    **
 *******************************************************************************

                              PM7 CALCULATION RESULTS

 *******************************************************************************
 *  CALCULATION DONE:                                Sat Sep 17 21:36:16 2022  *
 *  PM7        - The PM7 Hamiltonian to be used
 *  EF         - USE EF ROUTINE FOR MINIMUM SEARCH
 *  T=         - A TIME OF 172800.0

# arcファイルの読み取り

In [5]:
with open('phenol.arc', mode='r') as f:
    lines = f.readlines()
    for line in lines:
        print(line, end='')



                     SUMMARY OF  PM7 CALCULATION

                                                       MOPAC v22.0.4 MacOS
                                                       Sat Sep 17 21:36:16 2022

           Empirical Formula: C6 H6 O  =    13 atoms

 PM7 EF
 c1ccc(cc1)O
 InChI=1S/C6H6O/c7-6-4-2-1-3-5-6/h1-5,7H


     GEOMETRY OPTIMISED USING EIGENVECTOR FOLLOWING (EF).     
     SCF FIELD WAS ACHIEVED                                   

          HEAT OF FORMATION       =        -22.15781 KCAL/MOL =     -92.70830 KJ/MOL
          GRADIENT NORM           =          0.99656          =       0.27640 PER ATOM
          DIPOLE                  =          1.30764 DEBYE   POINT GROUP:  Cs  
          NO. OF FILLED LEVELS    =         18
          IONIZATION POTENTIAL    =          9.235653 EV
          HOMO LUMO ENERGIES (EV) =         -9.236  0.035
          MOLECULAR WEIGHT        =         94.1128
          COSMO AREA              =        129.57 SQUARE ANGSTROMS
          COSM

それぞれの意味

HEAT OF FORMATION       =        -22.15781 KCAL/MOL =     -92.70830 KJ/MOL
GRADIENT NORM           =          0.99656          =       0.27640 PER ATOM
DIPOLE                  =          1.30764 DEBYE   POINT GROUP:  Cs  
NO. OF FILLED LEVELS    =         18
IONIZATION POTENTIAL    =          9.235653 EV
HOMO LUMO ENERGIES (EV) =         -9.236  0.035
MOLECULAR WEIGHT        =         94.1128
COSMO AREA              =        129.57 SQUARE ANGSTROMS
COSMO VOLUME            =        119.27 CUBIC ANGSTROMS

# 自動で読み取って、pandasのデータフレームに入れるやり方